onnxruntime/contrib_ops/cpu/transformers/logits_processor.h

```diff
@@ -141,10 +150,125 @@ class PresencePenaltyLogitsProcessor : public ILogitsProcessor<T> {
 template <typename T>
 class TimestampLogitsProcessor : public ILogitsProcessor<T> {
  public:
-  TimestampLogitsProcessor(int eos_token_id, int max_initial_timestamp_index);
+  TimestampLogitsProcessor(int eos_token_id, int max_initial_timestamp_index)
+      : eos_token_id_(eos_token_id), max_initial_timestamp_index_(max_initial_timestamp_index) {}
 
   void Process(const ISequences* sequences,
-               NextTokenScores<T>& next_token_scores) override;
+               NextTokenScores<T>& next_token_scores) override {
+    const int beg_token_id_ = eos_token_id_ + 107;
+    const int not_token_id_ = eos_token_id_ + 106;
+    const int solm_token_id_ = eos_token_id_ + 105;
+    const int sot_token_id_ = eos_token_id_ + 1;
+    constexpr int translate_token_id_ = 50358;
+    constexpr int transcribe_token_id_ = 50359;
```

Comment:
The translate and transcribe token ids are different in the multilingual models (e.g. `openai/whisper-tiny`) vs English-only models (e.g. `openai/whisper-tiny.en`).

```
>>> from transformers import WhisperProcessor
>>> tiny = WhisperProcessor.from_pretrained("openai/whisper-tiny")
>>> tiny.get_decoder_prompt_ids(language="english", task="transcribe")
[(1, 50259), (2, 50359), (3, 50363)]
>>> tiny.get_decoder_prompt_ids(language="english", task="translate")
[(1, 50259), (2, 50358), (3, 50363)]
>>> tiny_en = WhisperProcessor.from_pretrained("openai/whisper-tiny.en")
>>> tiny_en.get_decoder_prompt_ids(language="english", task="transcribe")
[(1, 50258), (2, 50358), (3, 50362)]
>>> tiny_en.get_decoder_prompt_ids(language="english", task="translate")
[(1, 50258), (2, 50357), (3, 50362)]
```

Reply:
Nice finding! This is original implementation in ORT. Just move a place. Let's use different PR to fix this.